packages/fx-core/src/common/local/constants.ts

```diff
@@ -145,7 +145,7 @@ export const TaskDefaultValue = Object.freeze({
     },
   },
   npmInstall: {
-    npmInstallArgs: ["--no-audit"],
+    npmInstallArgs: ["--no-audit", "--legacy-peer-deps"],
```

Comment:
Will `--legacy-peer-deps` has a requirement for `npm` version?

Reply:
Both npm 6 and 8 can perform `--legacy-peer-deps`. Npm 6 would have warnings for peer dependency not installed. For instance, if users use `teamsfx-react@2.0.0` but `teamsfx@1.0.0`, it would have warnings.